sunkit_image/asda.py

```diff
@@ -11,410 +11,379 @@
-                for im in np.arange(-self.r, self.r + 1)
-                for jm in np.arange(-self.r, self.r + 1)
-            ],
-        )
-        return np.array([vel, vel - vel.mean(axis=0)])
-
-    def gamma_values(self):
-        """
-        Calculate ``gamma1`` and ``gamma2`` values of velocity field vx and vy.
-
-        Returns
-        -------
-        `tuple`
-            A tuple in form of ``(gamma1, gamma2)``, where ``gamma1`` is useful in
-            finding vortex centers and ``gamma2`` is useful in finding vortex
-            edges.
-        """
-        # This part of the code was written in (x, y) order
-        # but numpy is in (y, x) order so we need to transpose it
-        self.vx = self.vx.T
-        self.vy = self.vy.T
-        if self.factor > 1:
-            self.vx = reform2d(self.vx, self.factor)
-            self.vy = reform2d(self.vy, self.factor)
-        self.gamma = np.array([np.zeros_like(self.vx), np.zeros_like(self.vy)]).T
-        # pm vectors, see equation (8) in Graftieaux et al. 2001 or Equation (1) in Liu et al. 2019
-        pm = np.array(
-            [[i, j] for i in np.arange(-self.r, self.r + 1) for j in np.arange(-self.r, self.r + 1)],
-            dtype=float,
-        )
-        # Mode of vector pm
-        pnorm = np.linalg.norm(pm, axis=1)
-        # Number of points in the concerned region
-        N = (2 * self.r + 1) ** 2
-        index = np.array(
+    if vx.shape != vy.shape:
+        raise ValueError("Shape of velocity field's vx and vy do not match")
+    if not isinstance(r, int):
+        raise ValueError("Keyword 'r' must be an integer")
+    if not isinstance(factor, int):
+        raise ValueError("Keyword 'factor' must be an integer")
+
+    # This part of the code was written in (x, y) order
+    # but numpy is in (y, x) order so we need to transpose it
+    dshape = np.shape(vx)
+    vx = vx.T
+    vy = vy.T
+    if factor > 1:
+        vx = reform2d(vx, factor)
+        vy = reform2d(vy, factor)
+    gamma = np.array([np.zeros_like(vx), np.zeros_like(vy)]).T
+    # pm vectors, see equation (8) in Graftieaux et al. 2001 or Equation (1) in Liu et al. 2019
+    pm = np.array(
+        [[i, j] for i in np.arange(-r, r + 1) for j in np.arange(-r, r + 1)],
+        dtype=float,
+    )
+    # Mode of vector pm
+    pnorm = np.linalg.norm(pm, axis=1)
+    # Number of points in the concerned region
+    N = (2 * r + 1) ** 2
+
+    index = np.array(
+        [[i, j] for i in np.arange(r, dshape[0] - r) for j in np.arange(r, dshape[1] - r)],
+    )
+    index = index.T
+    vel = generate_velocity_field(vx, vy, index[1], index[0], r)
+    for d, (i, j) in enumerate(
+        product(np.arange(r, dshape[0] - r, 1), np.arange(r, dshape[1] - r, 1)),
+    ):
+        gamma[i, j, 0], gamma[i, j, 1] = calculate_gamma(pm, vel[..., d], pnorm, N)
+    # Transpose back vx & vy
+    vx = vx.T
+    vy = vy.T
+    return gamma
+
+
+def center_edge(gamma, rmin=4, gamma_min=0.89, factor=1):

```

Comment:
```suggestion
def get_vortex_edges(gamma, rmin=4, gamma_min=0.89, factor=1):
```
?

Reply:
Sure, I can change them if you'd like.

Reply:
Let's go with that

Reply:
should we also change rename the remaining function (```get_vtheta, get_vradial", get_vxvy,```)?

Reply:
Could do, but I don't have any sound ideas.

Reply:
get_vtheta -> get_rotational_velocity
get_vradial -> get_radial_velocity
get_vxvy -> get_velocity_field?

Reply:
Sounds good. Lets do it.